test-integration/repay-loan/1_repay_loan_3_payments.js

```diff
@@ -0,0 +1,176 @@
+    requestTime: currentTimestamp,
+    caller: loansInstance.address,
+    consensusAddress: loanTermConsensusInstance.address,
+  };
+  const loanResponseInfoTemplate = {
+    responseTime: currentTimestamp - 10,
+    interestRate: 400,
+    collateralRatio: 600,
+    maxLoanAmount: maxAmountWei.toFixed(0),
+    consensusAddress: loanTermConsensusInstance.address,
+  };
+  const loanTermsRequest = createLoanTermsRequest(loanTermsRequestInfo, chainId);
+  const signedResponses = await createMultipleSignedLoanTermsResponses(
+    web3,
+    loanTermsRequest,
+    signers,
+    loanResponseInfoTemplate,
+    nonces,
+    chainId,
+  );
+
+  const createLoanWithTermsResult = await loansInstance.createLoanWithTerms(
+    loanTermsRequest.loanTermsRequest,
+    signedResponses,
+    borrowerTxConfigWithValue.value,
+    borrowerTxConfigWithValue
+  );
+
+  const termsExpiryTime = await settingsInstance.termsExpiryTime();
+  const expiryTermsExpected = await timer.getCurrentTimestampInSecondsAndSum(termsExpiryTime);
+  const loanIDs = await loansInstance.getBorrowerLoans(borrower);
+  const lastLoanID = loanIDs[loanIDs.length - 1];
+  loans
+    .loanTermsSet(createLoanWithTermsResult)
+    .emitted(
+      lastLoanID,
+      borrowerTxConfigWithValue.from,
+      recipient,
+      loanResponseInfoTemplate.interestRate,
+      loanResponseInfoTemplate.collateralRatio,
+      loanResponseInfoTemplate.maxLoanAmount,
+      loanTermsRequestInfo.duration,
+      expiryTermsExpected,
+    );
+
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const nextTimestamp = await timer.getCurrentTimestampInSecondsAndSum(minutesToSeconds(2));
+  await timer.advanceBlockAtTime(nextTimestamp);
+  
+  // Take out a loan.
+  console.log(`Taking out loan id ${lastLoanID}...`);
+  const takeOutLoanResult = await loansInstance.takeOutLoan(lastLoanID, amountWei, borrowerTxConfig);
+  loans
+    .loanTakenOut(takeOutLoanResult)
+    .emitted(lastLoanID, borrowerTxConfig.from, amountWei);
+
+  // Set a lower price for Token/ETH.
+  console.log(`Settings final (lower) oracle price: 1 ${tokenName} = ${finalOraclePrice.toFixed(0)} WEI = ${toUnits(finalOraclePrice, 18)} ETHER`);
+  await chainlinkOracle.setLatestAnswer(finalOraclePrice);
+
+  // Calculate payment
+  console.log(`Making payment for loan id ${lastLoanID}...`);
+  const payment = toDecimals(34.67, decimals).toFixed(0);
+  console.log('Payment....', payment);
+
+  // Advance time to make first payment 10 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const paymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(10));
+  await timer.advanceBlockAtTime(paymentTimestamp);
+
+  // Make 1st payment
+  console.log(`Repaying loan id ${lastLoanID}...`);
+  console.log('Making 1st payment...');
+  await token.approve(loansInstance.address, payment, {from:borrower});
```

Comment:
I  think you don't need this approve call (to loansInstance).

Reply:
Removed

Reply:
The same. I think it wasn't removed.